solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2012,7 +2018,11 @@ private void checkStateInZk(CoreDescriptor cd) throws InterruptedException, NotI
         if (error == null)
           error = "coreNodeName " + coreNodeName + " does not exist in shard " + cloudDesc.getShardId() +
               ", ignore the exception if the replica was deleted";
-        throw new NotInClusterStateException(ErrorCode.SERVER_ERROR, error);
+        if (foundReplica.get() == null) {
+          throw new NotInClusterStateException(ErrorCode.SERVER_ERROR, error);
+        } else {
+          throw new InconsistentClusterStateException(ErrorCode.SERVER_ERROR, error, foundReplica.get());
```

Comment:
this exception is catched by caller. And caller will just log this exception?

Or should we catch this exception explicitly in caller?